### PR TITLE
CU/7059-Dylan-ClaimFileUploadAndTimeline

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Claim File Upload and Timeline Unit tests

## Screenshots/Video
N/A

## Testing
Yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
S src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ConsolidatedClaimsNote/ConsolidatedClaimsNote.test.tsx
S src/screens/BenefitsScreen/ClaimsScreen/NeedHelpData/NeedHelpData.test.tsx
S src/screens/BenefitsScreen/ClaimsScreen/ClaimLettersScreen/NoClaimLettersScreen/NoClaimLettersScreen.test.tsx
S src/screens/BenefitsScreen/ClaimsScreen/NoClaimsAndAppealsAccess/NoClaimsAndAppealsAccess.test.tsx
M src/screens/BenefitsScreen/ClaimsScreen/ClaimsAndAppealsListView/ClaimsAndAppealsListView.test.tsx
S src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimTimeline.test.tsx
M src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/PhaseIndicator.test.tsx
M src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimPhase.test.tsx

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by theodur — CHANGES_REQUESTED** [review]
Seeing same issues here
https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458

**Comment by Sparowhawk** [comment]
> Seeing same issues here [#7148 (review)](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458)

updated

**Review by alexandec — APPROVED** [review]

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx [hunk 1]
```diff
@@ -61,36 +52,36 @@ context('UploadFile', () => {
...
-        testInstance.findAllByType(VAButton)[0].props.onPress()
-      })
-
-      expect(findByTypeWithText(testInstance, TextView, 'Check the box to confirm the information is correct')).toBeTruthy()
+    it('should display an error if the checkbox is not checked', () => {
+      fireEvent.press(screen.getByTestId('Document type picker required'))
+      fireEvent.press(screen.getByTestId('Civilian Police Reports'))
+      fireEvent.press(screen.getByTestId('Done'))
+      fireEvent.press(screen.getByTestId('Submit file'))
+      expect(screen.getByText('Check the box to confirm the information is correct')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:75]:
```suggestion
      fireEvent.press(screen.getByRole('spinbutton', { name: 'Document type picker required' }))
      fireEvent.press(screen.getByRole('menuitem', { name: 'Civilian Police Reports' }))
      fireEvent.press(screen.getByRole('button', { name: 'Done' }))
      fireEvent.press(screen.getByRole('button', { name: 'Submit file' }))
      expect(screen.getByRole('checkbox', { name: 'Check the box to confirm the information is correct' })).toBeTruthy()
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx [hunk 2]
```diff
@@ -61,36 +52,36 @@ context('UploadFile', () => {
...
-      act(() => {
-        testInstance.findByType(VAModalPicker).props.onSelectionChange('L228')
-        testInstance.findByType(VASelector).props.onSelectionChange(true)
-        testInstance.findAllByType(VAButton)[0].props.onPress()
-      })
-
+    it('should bring up confirmation requirements are met', () => {
+      fireEvent.press(screen.getByTestId('Document type picker required'))
+      fireEvent.press(screen.getByTestId('Civilian Police Reports'))
+      fireEvent.press(screen.getByTestId('Done'))
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:82]:
```suggestion
      fireEvent.press(screen.getByRole('spinbutton', { name: 'Document type picker required' }))
      fireEvent.press(screen.getByRole('menuitem', { name: 'Civilian Police Reports' }))
      fireEvent.press(screen.getByRole('button', { name: 'Done' }))
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx [hunk 3]
```diff
@@ -61,36 +52,36 @@ context('UploadFile', () => {
...
+      fireEvent.press(screen.getByLabelText('The file I uploaded is evidence for this claim. (Required) '))
+      fireEvent.press(screen.getByTestId('Submit file'))
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:84]:
```suggestion
      fireEvent.press(screen.getByRole('button', { name: 'Submit file' }))
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.test.tsx [hunk 4]
```diff
@@ -19,18 +14,20 @@ context('TakePhotos', () => {
...
   }
 
-  beforeEach(() => {
+  it('initializes correctly', () => {
     initializeTestInstance()
-  })
-
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+    expect(screen.getByText('This feature is not yet accessible to screen readers')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.test.tsx:23]:
```suggestion
    expect(screen.getByRole('tab', { name: 'This feature is not yet accessible to screen readers' })).toBeTruthy()
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx [hunk 5]
```diff
@@ -52,15 +41,16 @@ context('UploadOrAddPhotos', () => {
...
     initializeTestInstance()
-  })
-
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+    expect(screen.getByRole('header', { name: 'Upload photos' })).toBeTruthy()
+    expect(screen.getByText('Add photo')).toBeTruthy()
+    expect(screen.getByText('of 10 photos')).toBeTruthy()
+    expect(screen.getByText('0 Bytes of 50MB')).toBeTruthy()
+    expect(screen.getByTestId('Document type picker required')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx:52]:
```suggestion
    expect(screen.getByRole('spinbutton', { name: 'Document type (Required)' })).toBeTruthy()
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx [hunk 6]
```diff
@@ -52,15 +41,16 @@ context('UploadOrAddPhotos', () => {
...
 
-  beforeEach(() => {
+  it('initializes correctly', () => {
     initializeTestInstance()
-  })
-
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+    expect(screen.getByRole('header', { name: 'Upload photos' })).toBeTruthy()
+    expect(screen.getByText('Add photo')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx:49]:
```suggestion
   expect(screen.getByRole('button', { name: 'Add photo' })).toBeTruthy()
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimPhase.test.tsx [hunk 7]
```diff
@@ -1,19 +1,14 @@
...
-import { claim } from '../../../claimData'
 import ClaimPhase from './ClaimPhase'
-import PhaseIndicator from './PhaseIndicator'
-import { TextView, VAButton, VAIcon } from 'components'
-import { Pressable } from 'react-native'
-import { waitFor } from '@testing-library/react-native'
 import { when } from 'jest-when'
 
+import { context, render } from 'testUtils'
+import { screen, fireEvent } from '@testing-library/react-native'
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimPhase.test.tsx:6]: Import can be grouped with the other node modules

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/AskForClaimDecision/AskForClaimDecision.test.tsx [hunk 8]
```diff
@@ -108,21 +95,16 @@ context('AskForClaimDecision', () => {
...
   describe('on click of submit', () => {
     describe('if the check box is not checked', () => {
-      it('should display the field error', async () => {
-        act(() => {
-          testInstance.findByType(VASelector).props.onSelectionChange(false)
-          testInstance.findByType(VAButton).props.onPress()
-        })
-
+      it('should display the field error', () => {
+        fireEvent.press(screen.getByText('Request claim evaluation'))
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/AskForClaimDecision/AskForClaimDecision.test.tsx:99]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:75]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx (hunk 3 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:84]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx:52]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx (hunk 6 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.test.tsx:49]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.test.tsx (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.test.tsx:23]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.test.tsx:82]: updated

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimPhase.test.tsx (hunk 7 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimPhase.test.tsx:6]: updated

**Review by alexandec — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**